tests/functional/object/mcg/test_internal_route_auth.py

```diff
@@ -15,13 +22,84 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.fixture(autouse=True)
+def reset_configmap(request):
+    """
+    resets noobaa metrics configmap values to original values
+    """
+    noobaa_configmap = OCP(
+        kind="configmap",
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        resource_name=constants.NOOBAA_CONFIGMAP,
+    )
+
+    def reset_flag():
+        params = '{{"data": {{"NOOBAA_VERSION_AUTH_ENABLED": "true"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
+        params = '{{"data": {{"NOOBAA_METRICS_AUTH_ENABLED": "true"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
+
+    request.addfinalizer(reset_flag)
+
+
 @mcg
 @red_squad
 class TestNoobaaRouteAuthentication:
     """
     Tests noobaa management route and noobaa endpoints with JWT authentication
     """
 
+    def change_config_flag(self, flag_param):
+        """
+        Changes flag value to false for testing purpose
+        """
+        noobaa_configmap = OCP(
+            kind="configmap",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name=constants.NOOBAA_CONFIGMAP,
+        )
+        params = f'{{"data": {{"{flag_param}": "false"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
```

Comment:
1. Please consider making this more flexible. It'll make the usage self explanatory (no need for the comment like the one in line 234), and you'll be able to use it to set other values for other test cases.
2. Use the `config.ENV_DATA["cluster_namespace"]` instead of `constants.OPENSHIFT_STORAGE_NAMESPACE`. IIRC using the existing one will fail the test when tested by teams outside of ODF-QE.

```suggestion
def set_nb_config_flag(self, flag_param, value):
    """
    Sets the value for a given config flag in noobaa-config.

    Args:
        flag_param (str): The config flag key to modify.
        value (str): Desired value for the flag.
    """
    noobaa_configmap = OCP(
        kind="configmap",
        namespace=config.ENV_DATA["cluster_namespace"],
        resource_name=constants.NOOBAA_CONFIGMAP,
    )
    params = f'{{"data": {{"{flag_param}": "{value}"}}}}'
    noobaa_configmap.patch(params=params, format_type="merge")
```

Reply:
Fixed